src/services/membership.js

```diff
@@ -148,23 +148,21 @@ const get = (activityCode, sessionCode) => {
 };
 
 //Change the privacy value for a club membership
-const toggleMembershipPrivacy = userMembership => {
+const toggleMembershipPrivacy = async userMembership => {
   let currentMembershipPrivacy = userMembership.Privacy;
   let newMembershipPrivacy = !currentMembershipPrivacy;
   let setMembershipPrivacy = async function(value) {
-    return http
+    return await http
       .put('memberships/' + userMembership.MembershipID + '/privacy/' + value, value)
-      .catch(reason => {
-        console.log(reason);
-        //TODO handle error
-      });
+      .then(() => {
+        // Changes the membership's privacy to the new privacy
+        userMembership.Privacy = newMembershipPrivacy;
+        return true;
+      })
+      .catch(false);
```

Comment:
I removed the console log because that was not needed. I also placed "userMembership.Privacy = newMembershipPrivacy" inside of the block where all changes are made if the HTTP PUT was successful. When that code statement was outside of the block, the privacy on the UI would change even if the HTTP PUT method failed.